compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/FirCallCompletionResultsWriterTransformer.kt

```diff
@@ -436,7 +437,29 @@ class FirCallCompletionResultsWriterTransformer(
         // Control flow info is necessary prerequisite because we collect return expressions in that function
         //
         // Example: second lambda in the call like list.filter({}, {})
-        if (!dataFlowAnalyzer.isThereControlFlowInfoForAnonymousFunction(anonymousFunction)) return anonymousFunction.compose()
+        if (!dataFlowAnalyzer.isThereControlFlowInfoForAnonymousFunction(anonymousFunction)) {
+            // But, don't leave implicit type refs behind
+            val implicitTypeTransformer = object : FirDefaultTransformer<Nothing?>() {
+                override fun <E : FirElement> transformElement(element: E, data: Nothing?): CompositeTransformResult<E> {
+                    @Suppress("UNCHECKED_CAST")
+                    return (element.transformChildren(this, data) as E).compose()
+                }
+
+                override fun transformImplicitTypeRef(
+                    implicitTypeRef: FirImplicitTypeRef,
+                    data: Nothing?
+                ): CompositeTransformResult<FirTypeRef> =
+                    buildErrorTypeRef {
+                        source = implicitTypeRef.source
+                        diagnostic = ConeSimpleDiagnostic("Cannot infer type w/o CFG", DiagnosticKind.InferenceError)
+                    }.compose()
+
+            }
+            // TODO: should we transform simply all children?
```

Comment:
Maybe, I'm not sure. You can try to test case when we have not-analyzed lambda with non-trivial body

Reply:
I already tried that, and saw too many redundant type error messages. Changed this as a note instead.